source/server/listener_impl.cc

```diff
@@ -280,7 +281,7 @@ ProcessContextOptRef ListenerFactoryContextBaseImpl::processContext() {
 }
 Configuration::ServerFactoryContext&
 ListenerFactoryContextBaseImpl::getServerFactoryContext() const {
```

Comment:
Don't we already have a context wrapper in here that could do the wrapping? There are so many wrappers. :joy:

Reply:
given this is a new class I don't think so?

Reply:
I guess the larger point is there are so many wrappers it's incredibly confusing. One general thought: At some point it made sense (I guess) to have all of the inheritance. Does that makes sense anymore? Would it be cleaner / easier to understand if all the factory interfaces were split? Not sure just throwing it out there.